sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -398,133 +384,117 @@ export class EventHubSender extends LinkEntity {
+              });
+            },
+            abortSignal: options?.abortSignal,
+            timeoutMs: timeoutInMs,
+            timeoutMessage:
+              `[${this._context.connectionId}] Sender "${this.name}" ` +
+              `with address "${this.address}", was not able to send the message right now, due ` +
+              `to operation timeout.`
+          });
+        } catch (err) {
+          const translatedError = translate(err);
+          logger.warning(
+            "[%s] An error occurred while creating the sender %s: %s",
+            this._context.connectionId,
+            this.name,
+            `${translatedError?.name}: ${translatedError?.message}`
+          );
+          logErrorStackTrace(translatedError);
+          throw translatedError;
         }
+      }
+      const timeTakenByInit = Date.now() - initStartTime;
 
-        const actionAfterTimeout = (): void => {
-          removeListeners();
-          const desc: string =
-            `[${this._context.connectionId}] Sender "${this.name}" with ` +
-            `address "${this.address}", was not able to send the message right now, due ` +
-            `to operation timeout.`;
-          logger.warning(desc);
-          const e: Error = {
-            name: "OperationTimeoutError",
-            message: desc
-          };
-          return reject(translate(e));
-        };
+      logger.verbose(
+        "[%s] Sender '%s', credit: %d available: %d",
+        this._context.connectionId,
+        this.name,
+        this._sender!.credit,
+        this._sender!.session.outgoing.available()
+      );
 
-        const waitTimer = setTimeout(actionAfterTimeout, timeoutInMs);
-        const initStartTime = Date.now();
-        if (!this.isOpen()) {
-          logger.verbose(
-            "Acquiring lock %s for initializing the session, sender and " +
-              "possibly the connection.",
-            this.senderLock
-          );
+      if (!this._sender!.sendable()) {
+        logger.verbose(
+          "%s Sender '%s', waiting for 1 second for sender to become sendable",
+          this._context.connectionId,
+          this.name
+        );
 
-          try {
-            const senderOptions = this._createSenderOptions(timeoutInMs);
-            await defaultLock.acquire(this.senderLock, () => {
-              return this._init(senderOptions);
-            });
-          } catch (err) {
-            removeListeners();
-            const translatedError = translate(err);
-            logger.warning(
-              "[%s] An error occurred while creating the sender %s: %s",
-              this._context.connectionId,
-              this.name,
-              `${translatedError?.name}: ${translatedError?.message}`
-            );
-            logErrorStackTrace(translatedError);
-            return reject(translatedError);
-          }
-        }
-        const timeTakenByInit = Date.now() - initStartTime;
+        await delay(1000);
```

Comment:
We had this in SB, I don't see why the same reasons would not apply here. So, added it to keep the code similar between the two projects to help in future consolidation

Reply:
Should we set the delay to `timeoutInMs - timeTakenByInit`?

Someday I'd like to wrap the event handler that indicates that messages are sendable in a promise and do a race between this timeout and that promise, then we wouldn't need the 2nd sendable() check.

Reply:
> Should we set the delay to timeoutInMs - timeTakenByInit

Or Min(1000, timeoutInMs - timeTakenByInit)?
Or not await at all if timeoutInMs - timeTakenByInit < 1000 ?

> Someday I'd like to wrap the event handler that indicates that messages are sendable in a promise and do a race between this timeout and that promise, then we wouldn't need the 2nd sendable() check.

Something like #4764?

Reply:
> Or not await at all if timeoutInMs - timeTakenByInit < 1000 ?

I think that seems reasonable.

> Something like #4764?

Exactly!

Reply:
Done in https://github.com/Azure/azure-sdk-for-js/pull/13826/commits/cf367c28fc4481db6f4ef7e102827f7220fd1610